autopts/ptsprojects/stack/layers/pbp.py

```diff
@@ -31,3 +33,15 @@ def wait_public_broadcast_event_found_ev(self, addr_type, addr, broadcast_name,
             self.event_queues[defs.BTP_PBP_EV_PUBLIC_BROADCAST_ANNOUNCEMENT_FOUND],
             lambda ev: (addr_type, addr, broadcast_name) == (ev['addr_type'], ev['addr'], ev['broadcast_name']),
             timeout, remove)
+
+    def set_program_info(self, program_info):
+        if isinstance(program_info, str) and len(program_info > 0):
+            self.program_info = program_info
+        else:
+            raise Exception('Tried to set improper program info')
+
+    def set_broadcast_name(self, broadcast_name):
+        if isinstance(broadcast_name, str) and len(broadcast_name >= 4) and len(broadcast_name <= 32):
+            self.broadcast_name = broadcast_name
+        else:
+            raise Exception('Tried to set improper broadcast name')
```

Comment:
Just to make this a bit more explicit: 

If the `isinstance(broadcast_name, str)` check fails, we should raise a TypeError, and if the length check fails, we should raise a ValueError. 

We should not rely on the basic Exception

Reply:
Fixed

Reply:
Done